src/components/lists/CardList.js

```diff
@@ -0,0 +1,82 @@
+import EasyCards from './EasyCards';
+import MediumCards from './MediumCards';
+import DifficultCards from './DifficultCards';
+import AllCards from './AllCards';
+
+export default function CardList({
+  onDeleteConfirm,
+  onTrashClick,
+  onKeepConfirm,
+  showModal,
+  onPinClick,
+  onCountRights,
+  onCountWrongs,
+  easyCards,
+  mediumCards,
+  difficultCards,
+  onShowHide,
+  difficulty,
+  cards,
+  value,
+}) {
+  if (difficulty === 'easy')
+    return (
+      <EasyCards
+        easyCards={easyCards}
+        showModal={showModal}
+        onPinClick={onPinClick}
+        onCountRights={onCountRights}
+        onCountWrongs={onCountWrongs}
+        onDeleteConfirm={onDeleteConfirm}
+        onKeepConfirm={onKeepConfirm}
+        onTrashClick={onTrashClick}
+        onShowHide={onShowHide}
+        value={value}
+      />
+    );
+  else if (difficulty === 'medium')
+    return (
+      <MediumCards
+        mediumCards={mediumCards}
+        showModal={showModal}
+        onPinClick={onPinClick}
+        onCountRights={onCountRights}
+        onCountWrongs={onCountWrongs}
+        onDeleteConfirm={onDeleteConfirm}
+        onKeepConfirm={onKeepConfirm}
+        onTrashClick={onTrashClick}
+        onShowHide={onShowHide}
+        value={value}
+      />
+    );
+  else if (difficulty === 'difficult')
+    return (
+      <DifficultCards
+        difficultCards={difficultCards}
+        showModal={showModal}
+        onPinClick={onPinClick}
+        onCountRights={onCountRights}
+        onCountWrongs={onCountWrongs}
+        onDeleteConfirm={onDeleteConfirm}
+        onKeepConfirm={onKeepConfirm}
+        onTrashClick={onTrashClick}
+        onShowHide={onShowHide}
+        value={value}
+      />
+    );
+  else
+    return (
+      <AllCards
+        cards={cards}
+        showModal={showModal}
+        onPinClick={onPinClick}
+        onCountRights={onCountRights}
+        onCountWrongs={onCountWrongs}
+        onDeleteConfirm={onDeleteConfirm}
+        onKeepConfirm={onKeepConfirm}
+        onTrashClick={onTrashClick}
+        onShowHide={onShowHide}
+        value={value}
+      />
+    );
+}
```

Comment:
As far as I can see, all of these components are literally the same. You just need to pass different props to `cards`. Please recheck and see whether you can just have one component `Cards` instead of `EasyCards`, `MediumCards`, `DifficultCards` and `AllCards`.